akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -257,27 +267,35 @@ final class ClusterClient(settings: ClusterClientSettings) extends Actor with Ac
   def receive = establishing
 
   def establishing: Actor.Receive = {
-    case Contacts(contactPoints) ⇒
-      if (contactPoints.nonEmpty) {
-        contacts = contactPoints.map(context.actorSelection)
-        contacts foreach { _ ! Identify(None) }
-      }
-    case ActorIdentity(_, Some(receptionist)) ⇒
-      log.info("Connected to [{}]", receptionist.path)
-      scheduleRefreshContactsTick(refreshContactsInterval)
-      sendBuffered(receptionist)
-      context.become(active(receptionist))
-      failureDetector.heartbeat()
-    case ActorIdentity(_, None) ⇒ // ok, use another instead
-    case HeartbeatTick ⇒
-      failureDetector.heartbeat()
-    case RefreshContactsTick ⇒ sendGetContacts()
-    case Send(path, msg, localAffinity) ⇒
-      buffer(DistributedPubSubMediator.Send(path, msg, localAffinity))
-    case SendToAll(path, msg) ⇒
-      buffer(DistributedPubSubMediator.SendToAll(path, msg))
-    case Publish(topic, msg) ⇒
-      buffer(DistributedPubSubMediator.Publish(topic, msg))
+    val cancelable = context.system.scheduler.scheduleOnce(settings.reconnectTimeout, self, ReconnectFailureTick)
+
+    {
+      case Contacts(contactPoints) ⇒
+        if (contactPoints.nonEmpty) {
+          contacts = contactPoints.map(context.actorSelection)
+          contacts foreach { _ ! Identify(None) }
+        }
+      case ActorIdentity(_, Some(receptionist)) ⇒
+        log.info("Connected to [{}]", receptionist.path)
+        scheduleRefreshContactsTick(refreshContactsInterval)
+        sendBuffered(receptionist)
+        context.become(active(receptionist))
+        cancelable.cancel()
+        failureDetector.heartbeat()
+      case ActorIdentity(_, None) ⇒ // ok, use another instead
+      case HeartbeatTick ⇒
+        failureDetector.heartbeat()
+      case RefreshContactsTick ⇒ sendGetContacts()
+      case Send(path, msg, localAffinity) ⇒
+        buffer(DistributedPubSubMediator.Send(path, msg, localAffinity))
+      case SendToAll(path, msg) ⇒
+        buffer(DistributedPubSubMediator.SendToAll(path, msg))
+      case Publish(topic, msg) ⇒
+        buffer(DistributedPubSubMediator.Publish(topic, msg))
+      case ReconnectFailureTick ⇒
+        log.info(s"Receptionist reconnect not successful within ${settings.reconnectTimeout.toSeconds}s stopping cluster client")
```

Comment:
Was my first thought as well, but then I looked at other "connection loss" in this class and they use info. Although, I guess they aren't as severe. Will change.